internal/server/iigointernal/judiciary.go

```diff
@@ -310,3 +338,75 @@ func removeSanctions(slice []roles.Sanction, s int) []roles.Sanction {
 func getDifferenceInLength(slice1 []roles.Sanction, slice2 []roles.Sanction) int {
 	return len(slice1) - len(slice2)
 }
+
+// pickUpRulesByVariable returns a list of rule_id's which are affected by certain variables.
+func pickUpRulesByVariable(variableName rules.VariableFieldName, ruleStore map[string]rules.RuleMatrix, variableMap map[rules.VariableFieldName]rules.VariableValuePair) ([]string, bool) {
+	var Rules []string
+	if _, ok := variableMap[variableName]; ok {
+		for k, v := range ruleStore {
+			_, found := searchForVariableInArray(variableName, v.RequiredVariables)
+			if found {
+				Rules = append(Rules, k)
+			}
+		}
+		return Rules, true
+	} else {
+		// fmt.Sprintf("Variable name '%v' was not found in the variable cache", variableName)
+		return []string{}, false
+	}
+}
+
+func searchForVariableInArray(val rules.VariableFieldName, array []rules.VariableFieldName) (int, bool) {
+	for i, v := range array {
+		if v == val {
+			return i, true
+		}
+	}
+	return -1, false
+}
+
+// cullCheckedRules removes any entries in the history that have been evaluated in evalResults (for historical retribution)
+func cullCheckedRules(iigoHistory []shared.Accountability, evalResults map[shared.ClientID]roles.EvaluationReturn, rulesCache map[string]rules.RuleMatrix, variableCache map[rules.VariableFieldName]rules.VariableValuePair) []shared.Accountability {
```

Comment:
So while do think this function works as you want, it would be fantastic to under why do these things. Currently I'm really lost as to why these functions would be of use to a judge. Is there going to be an accompanying document tell teams how the system works or is there no need to understand why this happens.

Reply:
Well I don't mind writing something that explains that function, it won't be exposed to the judge at all, it's part of IIGO's internal auditing system. But I admit it is a little complicated, but kinda needs to be for what it does.